checkov/terraform/checks/data/aws/GithubActionsOIDCTrustPolicy.py

```diff
@@ -4,54 +4,82 @@
 from checkov.common.util.type_forcers import force_list
 from checkov.terraform.checks.data.base_check import BaseDataCheck
 
-gh_repo_regex = re.compile(r'repo:[^/]+/[^/]+')
+gh_repo_regex = re.compile(r"[\w]+/.+")
+gh_abusable_claims = ["workflow", "environment", "ref", "context", "head_ref", "base_ref"]
 
 
 class GithubActionsOIDCTrustPolicy(BaseDataCheck):
     def __init__(self):
-        name = 'Ensure GitHub Actions OIDC trust policies only allows actions from a specific known organization'
+        name = "Ensure GitHub Actions OIDC authorization policies only allows safe claims and claim order"
```

Comment:
```suggestion
        name = "Ensure GitHub Actions OIDC authorization policies only allow safe claims and claim order"
```

Also, internal note: we'll need to update the descriptive name as well